### PR TITLE
Fix !include tag loading

### LINKED ISSUE #1113: !include constructor is broken on python 3.6
# Description

After adding pyyaml validation, the `!include` constructor stopped working. I am still unsure if the issue is due to yamale and how it handles nested yaml files.

We need to fix the include constructor so we can keep the functionality of including a yaml file inside another.

## Steps to Reproduce
Add [skill-chat](https://github.com/FabioRosado/skill-chat) to opsdroid
set up skill-chat with a costumised yaml - follow the readme
start opsdroid

## Expected Functionality
Opsdroid should be able to include the custom file inside the configuration

## Experienced Functionality
```
python -m opsdroid start
could not determine a constructor for the tag '!include'
  in "/Users/fabiorosado/Library/Application Support/opsdroid/configuration.yaml", line 130, column 16
```

## Versions
- **Opsdroid version:**
- **Python version:**
- **OS/Docker version:**

## Configuration File
Please include your version of the configuration file bellow.

```yaml
Skills:
  - name: chat
    path: /Users/fabiorosado/Documents/GitHub/skill-chat
    customise: !include customise.yaml

```

## Additional Details
Any other details you wish to include such as screenshots, console messages, etc.

### PR DESCRIPTION
**Pull request by MrNaif2018**
# Description

This issue is the same as by yaml/pyyaml#266.
To fix this all is needed is to replace `yaml.safe_load` with `yaml.oad(Loader=yaml.SafeLoader)`

Fixes #1113 

## Status
**READY**

## Type of change

_Please delete options that are not relevant._

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

From running the configuration yaml file is loaded fine.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by MrNaif2018** [comment]
I ran python -m opsdroid start on my end and it worked.
I guess you didn't update correctly.
Will retest again, but I think it works

**Comment by FabioRosado** [comment]
Are you using python 3.7 by any chance? Because the tests pass on python 3.7 but fail on 3.6

**Comment by MrNaif2018** [comment]
I am using 3.6

**Comment by FabioRosado** [comment]
Can you remove the unittest.skip like I said earlier and then run tox? Also check with a skill that uses the !include constructor and let me know what you find. It's odd that on your end works fine but mine is all broken haha

**Comment by MrNaif2018** [comment]
@FabioRosado Now it works.
The actual issue was that yamale was using CSafeLoader if it is available, so because of that tag wasn't registered. Now loader picks a loader the same way as yamale does and uses it for loading.

**Comment by FabioRosado** [comment]
I've added your fork to my machine and pull all these changes. I've tried to run opsdroid and this is what I get:
```
python -m opsdroid start
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/fabiorosado/Documents/GitHub/opsdroid/opsdroid/__main__.py", line 12, in <module>
    init()
  File "/Users/fabiorosado/Documents/GitHub/opsdroid/opsdroid/__main__.py", line 9, in init
    opsdroid.cli.cli()
  File "/Users/fabiorosado/.local/share/virtualenvs/opsdroid-13bLHlYD/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/fabiorosado/.local/share/virtualenvs/opsdroid-13bLHlYD/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/fabiorosado/.local/share/virtualenvs/opsdroid-13bLHlYD/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/fabiorosado/.local/share/virtualenvs/opsdroid-13bLHlYD/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/fabiorosado/.local/share/virtualenvs/opsdroid-13bLHlYD/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/fabiorosado/Documents/GitHub/opsdroid/opsdroid/cli/start.py", line 26, in start
    configure_lang(config)
  File "/Users/fabiorosado/Documents/GitHub/opsdroid/opsdroid/cli/utils.py", line 68, in configure_lang
    lang_code = config.get("lang", DEFAULT_LANGUAGE)
AttributeError: 'NoneType' object has no attribute 'get'
```

When I run tox without the include constructor tests being skipped the test fails as well with this:

```
        cls.loader.add_constructor("!include", include_constructor)
        try:
            with open(config_path, "r") as stream:
                _LOGGER.info(_("Loaded config from %s."), config_path)
                schema_path = os.path.abspath("opsdroid/configuration/schema.yaml")
                schema = yamale.make_schema(schema_path)
                data = yamale.make_data(config_path)
                yamale.validate(schema, data)
                return yaml.load(stream, Loader=cls.loader)

        except ValueError as error:
            _LOGGER.critical(error)
>           sys.exit(1)
E           SystemExit: 1

opsdroid/loader.py:375: SystemExit
----------------------------------------- Captured log call -----------------------------------------
CRITICAL opsdroid.loader:loader.py:374
Error validating data /Users/fabiorosado/Documents/GitHub/opsdroid/tests/configs/minimal_with_include.yaml with schema /Users/fabiorosado/Documents/GitHub/opsdroid/opsdroid/configuration/schema.yaml
	 : 'None' is not a map
```

The test.main is also breaking with these changes:

```
_________________________________________ TestCLI.test_main _________________________________________

self = <test_cli.TestCLI testMethod=test_main>

    def test_main(self):
        with pytest.warns(DeprecationWarning, match=".*opsdroid start.*"):
            runner = CliRunner()
            with mock.patch.object(OpsDroid, "run") as mock_run:
                runner.invoke(opsdroid.cli.cli, [])
>               assert mock_run.called
E               AssertionError: assert False
E                +  where False = <MagicMock name='run' id='4464650952'>.called

tests/test_cli.py:225: AssertionError
```

Yesterday I have updated this PR but travis kept hanging so I'm going to assume it's an issue with the tests failing.

I'm going to update this branch once more and see if travis runs if not I'm going to say that the PR doesn't fix the issue unfortunately.

**Comment by FabioRosado** [comment]
I've played around with this and it seems that the issue is when you assign `loader` to one of the yaml loaders. We are using the `loader` to load the config, so I changed this variable name and the include constructor works, but the envvar constructor fails.

File: opsdroid/loader.py [hunk 1]
```diff
@@ -43,6 +43,11 @@
 class Loader:
     """Class to load in config and modules."""
 
+    try:
+        loader = yaml.CSafeLoader
```
Comment (FabioRosado) [inline opsdroid/loader.py]: Replace the loader variable with `yaml_loader`

File: opsdroid/loader.py [hunk 2]
```diff
@@ -338,7 +343,7 @@ def load_config_file(cls, config_paths):
             config_path = cls.create_default_config(DEFAULT_CONFIG_PATH)
 
         env_var_pattern = re.compile(r"^\$([A-Z_]*)$")
-        yaml.SafeLoader.add_implicit_resolver("!envvar", env_var_pattern, first="$")
+        cls.loader.add_implicit_resolver("!envvar", env_var_pattern, first="$")
```
Comment (FabioRosado) [inline opsdroid/loader.py]: Update the variable name after you have change it with `yaml_loader`

File: opsdroid/loader.py [hunk 3]
```diff
@@ -352,19 +357,18 @@ def include_constructor(loader, node):
             included_yaml = os.path.join(main_yaml_path, loader.construct_scalar(node))
 
             with open(included_yaml, "r") as included:
-                return yaml.safe_load(included)
-
-        yaml.SafeLoader.add_constructor("!envvar", envvar_constructor)
-        yaml.SafeLoader.add_constructor("!include", include_constructor)
+                return yaml.load(stream, Loader=cls.loader)
```
Comment (FabioRosado) [inline opsdroid/loader.py]: Here you need to load the included file not the stream, please replace this with:
`return yaml.load(included, Loader=cls.yaml_loader)`

**Comment by MrNaif2018** [comment]
@FabioRosado I fixed it

**Comment by MrNaif2018** [comment]
@FabioRosado I am not sure but I get 100% test coverage when running with tox. So not sure what to fix. Attribute error seems already covered

**Comment by jacobtomlinson** [comment]
I'm happy for this to be merged now. I wouldn't stress too much about the coverage, codecov has been acting up lately.

Let's get it onto master and see how the coverage does there.

@FabioRosado can you just check you are happy with everything?

**Comment by FabioRosado** [comment]
Yeah all good I mean the coverage is down on the loader but overall is higher so its good haha

@MrNaif2018 I'm going to merge this PR since travis is still acting up on your own. If you would like to take a stab and trying to cover the try/excerpt feel free to raise a PR since it will count for your hacktoberfest as well 😄 👍 

Finally, thank you so much for helping sorting this out and making sure that everything works fine on our constructors 😄 We love to thank contributors by sending them [opsdroid stickers](https://medium.com/opsdroid/contributor-sticker-packs-738058ceda59)! Fill in the [claim form](https://forms.gle/iQZV9DjkaUiSaDXK6) to get yours!

**Pull request merged**